pkg/unittest/test_job.go

```diff
@@ -444,7 +442,7 @@ func SplitManifests(renderedManifests *bytes.Buffer) map[string]string {
 
 		match := re.FindStringSubmatch(block)
 
-		if match == nil || len(match) < 2 {
```

Comment:
Can we guarantee there will be a match?
According to the regex documentation no match result in a nil.

Reply:
Rolled back, as this is just an improvement, and outside of a scope.

I'm not sure when `match == nil` could be null.

Example https://go.dev/play/p/32SyXRtE_GO

And FindStringSubmatch without match returns empty slice https://github.com/golang/go/blob/391dde29a37f3fd450f7d61e3f220930e0164b89/src/regexp/regexp.go#L1038